### ISSUE TITLE
Verification Email | Lambda Function

### ISSUE THREAD

**Issue by aanchan** [aanchan/synap-epics#63]
A new template was made for email verification. The current email template looks bare. The template was designed in adobe XD. The template contained a lot of code that was not needed. The current template which is primarily HTML and CSS is attached to this.[https://app.zenhub.com/files/165302750/74cf3d08-ccb8-4d16-8c25-3b6e0012985d/download](https://app.zenhub.com/files/165302750/74cf3d08-ccb8-4d16-8c25-3b6e0012985d/download). The template looks something like this. 
![image (9).png](https://images.zenhubusercontent.com/5c363aebf6ef6263f5aa92c3/ae81c551-14e6-4fd7-8d0e-5e8f854cba92). 

The HTML template comes with a number of static assets. An [S3 bucket](https://s3.console.aws.amazon.com/s3/buckets/synap-confirmation-email/?region=ca-central-1) was created to hold these static assets with a cloudfront distribution (d3mseuj3wyzyi0.cloudfront.net) made to point to content.synaptitudebrainhealth.com. 

The equivalent HTML file is something like this :

```<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=edge"/>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Confirm Email Template</title>
<style id="applicationStylesheet" type="text/css">
	.mediaViewInfo {
		--web-view-name: Confirm Email Template;
		--web-view-id: Confirm_Email_Template;
		--web-scale-on-resize: true;
		--web-enable-deep-linking: true;
	}
	:root {
		--web-view-ids: Confirm_Email_Template;
	}
	* {
		margin: 0;
		padding: 0;
		box-sizing: border-box;
		border: none;
	}
	#Confirm_Email_Template {
		position: absolute;
		width: 800px;
		height: 1636px;
		background-color: rgba(255,255,255,1);
		overflow: hidden;
		--web-view-name: Confirm Email Template;
		--web-view-id: Confirm_Email_Template;
		--web-scale-on-resize: true;
		--web-enable-deep-linking: true;
	}
	#Welcome_to_the_Synaptitude_Bra {
		position: absolute;
		left: 139px;
		top: 239px;
		overflow: visible;
		width: 523px;
		white-space: nowrap;
		line-height: 60px;
		margin-top: -11px;
		text-align: center;
		font-family: Public Sans;
		font-style: normal;
		font-weight: bold;
		font-size: 38px;
		color: rgba(53,64,82,1);
	}
	#Group_26 {
		position: absolute;
		width: 406px;
		height: 74px;
		left: 197px;
		top: 982px;
		overflow: visible;
	}
	.Rectangle_2 {
		position: absolute;
		overflow: visible;
		width: 406px;
		height: 74px;
		left: 0px;
		top: 0px;
	}
	#Confirm_email_address {
		position: absolute;
		left: 70px;
		top: 20px;
		overflow: visible;
		width: 265px;
		white-space: nowrap;
		text-align: center;
		font-family: Roboto;
		font-style: normal;
		font-weight: bold;
		font-size: 26px;
		color: rgba(255,255,255,1);
	}
	#To_complete_your_registration_ {
		position: absolute;
		left: 109px;
		top: 813px;
		overflow: visible;
		width: 583px;
		height: 156px;
		text-align: left;
		font-family: Roboto;
		font-style: normal;
		font-weight: normal;
		font-size: 30px;
		color: rgba(53,64,82,1);
	}
	#Best_wishes__Your_brain_health {
		position: absolute;
		left: 109px;
		top: 1175px;
		overflow: visible;
		width: 306px;
		white-space: nowrap;
		line-height: 50px;
		margin-top: -10px;
		text-align: left;
		font-family: Roboto;
		font-style: normal;
		font-weight: bold;
		font-size: 30px;
		color: rgba(53,64,82,1);
	}
	#download {
		position: absolute;
		width: 137px;
		height: 93px;
		left: 332px;
		top: 62px;
		overflow: visible;
	}
	#Group_27 {
		position: absolute;
		width: 592px;
		height: 367px;
		left: 104px;
		top: 395px;
		overflow: visible;
	}
	.Rectangle_1 {
		position: absolute;
		overflow: visible;
		width: 592px;
		height: 356px;
		left: 0px;
		top: 8px;
	}
	#ID1 {
		position: absolute;
		width: 443px;
		height: 367px;
		left: 75px;
		top: 0px;
		overflow: visible;
	}
	#Group_31 {
		position: absolute;
		width: 750px;
		height: 258px;
		left: 25px;
		top: 1378px;
		overflow: visible;
	}
	.Rectangle_15 {
		position: absolute;
		overflow: visible;
		width: 750px;
		height: 258px;
		left: 0px;
		top: 0px;
	}
	#Group_29 {
		position: absolute;
		width: 712px;
		height: 228px;
		left: 19px;
		top: 24px;
		overflow: visible;
	}
	#Footer {
		position: absolute;
		width: 712px;
		height: 228px;
		left: 0px;
		top: 0px;
		overflow: visible;
	}
	#Copyright___2020_Synaptitude_B {
		position: absolute;
		left: 0px;
		top: 167px;
		overflow: visible;
		width: 713px;
		height: 61px;
		line-height: 25px;
		margin-top: -4.5px;
		text-align: center;
		font-family: Roboto;
		font-style: normal;
		font-weight: normal;
		font-size: 16px;
		color: rgba(53,64,82,1);
	}
	#Group_30 {
		position: absolute;
		width: 534px;
		height: 137px;
		left: 89px;
		top: 0px;
		overflow: visible;
	}
	#We_d_love_to_hear_from_you__Fo {
		position: absolute;
		left: 0px;
		top: 0px;
		overflow: visible;
		width: 535px;
		height: 137px;
		text-align: center;
		font-family: Roboto;
		font-style: normal;
		font-weight: bold;
		font-size: 30px;
		color: rgba(72,78,88,1);
	}
	#Symbol_1___3 {
		position: absolute;
		width: 11.339px;
		height: 21.834px;
		left: 306.492px;
		top: 123.375px;
		overflow: visible;
	}
	.facebook_2 {
		overflow: visible;
		position: absolute;
		width: 11.339px;
		height: 21.834px;
		left: 0px;
		top: 0px;
		transform: matrix(1,0,0,1,0,0);
	}
	#Rectangle_22 {
		position: absolute;
		width: 24px;
		height: 20.571px;
		left: 355.535px;
		top: 125.873px;
		overflow: visible;
	}
</style>

</head>
<body>
<div id="Confirm_Email_Template">
	<div id="Welcome_to_the_Synaptitude_Bra">
		<span>Welcome to the Synaptitude <br/>Brain Health Platform!</span>
	</div>
	<a href="http://www.synaptitudebrainhealth.com">
		<div id="Group_26">
			<svg class="Rectangle_2">
				<rect fill="rgba(93,49,201,1)" id="Rectangle_2" rx="30" ry="30" x="0" y="0" width="406" height="74">
				</rect>
			</svg>
			<div id="Confirm_email_address">
				<span>Confirm email address</span>
			</div>
		</div>
	</a>
	<div id="To_complete_your_registration_">
		<span>To complete your registration, confirm your email address with the button below: <br/></span><br>
	</div>
	<div id="Best_wishes__Your_brain_health">
		<span>Best wishes,<br/>Your brain health team</span><br>
	</div>
	<img id="download" src="http://content.synaptitudebrainhealth.com/download.png" srcset="http://content.synaptitudebrainhealth.com/download.png 1x, http://content.synaptitudebrainhealth.com/download@2x.png 2x">
	<div id="Group_27">
		<svg class="Rectangle_1">
			<rect fill="rgba(245,233,243,1)" id="Rectangle_1" rx="0" ry="0" x="0" y="0" width="592" height="356">
			</rect>
		</svg>
		<img id="ID1" src="http://content.synaptitudebrainhealth.com/ID1.png" srcset="http://content.synaptitudebrainhealth.com/ID1.png 1x, http://content.synaptitudebrainhealth.com/ID1@2x.png 2x">
	</div>
	<div id="Group_31">
		<svg class="Rectangle_15">
			<rect fill="rgba(245,245,245,1)" id="Rectangle_15" rx="0" ry="0" x="0" y="0" width="750" height="258">
			</rect>
		</svg>
		<div id="Group_29">
			<div id="Footer" class="Footer">
				<div id="Copyright___2020_Synaptitude_B">
					<span>Copyright © 2020</span><span style="font-style:normal;font-weight:bold;"> Synaptitude Brain Health</span><span>. All Rights Reserved.<br/>communications@synaptitude.info | www.synaptitudebrainhealth.com</span>
				</div>
				<div id="Group_30">
					<div id="We_d_love_to_hear_from_you__Fo">
						<span>We'd love to hear from you!</span><br><span style="font-style:normal;font-weight:lighter;font-size:20px;color:rgba(53,64,82,1);">Follow us on our Facebook Page or leave us a message. <br/>We are constantly improving and welcome feedback.</span>
					</div>
				</div>
				<div id="Symbol_1___3" class="Symbol_1___3">
					<svg class="facebook_2" viewBox="88.428 12.829 11.339 21.834">
						<path fill="rgba(53,64,82,1)" id="facebook_2" d="M 95.78785705566406 34.66301727294922 L 95.78785705566406 24.70341873168945 L 99.13092041015625 24.70341873168945 L 99.63143157958984 20.82200241088867 L 95.78785705566406 20.82200241088867 L 95.78785705566406 18.34383392333984 C 95.78785705566406 17.22009658813477 96.09995269775391 16.45420837402344 97.7115478515625 16.45420837402344 L 99.76692199707031 16.4532585144043 L 99.76692199707031 12.98177623748779 C 99.41127777099609 12.93464660644531 98.19127655029297 12.82899951934814 96.77189636230469 12.82899951934814 C 93.80850219726563 12.82899951934814 91.77970123291016 14.63775539398193 91.77970123291016 17.95962524414063 L 91.77970123291016 20.82210922241211 L 88.42799377441406 20.82210922241211 L 88.42799377441406 24.70352554321289 L 91.77960205078125 24.70352554321289 L 91.77960205078125 34.66312408447266 L 95.78785705566406 34.66301345825195 Z">
						</path>
					</svg>
				</div>
			</div>
			<img id="Rectangle_22" src="http://content.synaptitudebrainhealth.com/Rectangle_22.png" srcset="http://content.synaptitudebrainhealth.com/Rectangle_22.png 1x, http://content.synaptitudebrainhealth.com/Rectangle_22@2x.png 2x">
		</div>
	</div>
</div>
</body>
</html> 
```

Currently this HTML code being used in a US-East-1 Cognito Lambda function called [`Cognito-Verification-Email`](https://console.aws.amazon.com/lambda/home?region=us-east-1#/functions/Cognito-Verification-Email?tab=configuration). The email when attached to a user pool with a cognito-sync-trigger seems to fail. 

I get an email like this.

![Selection_706.png](https://images.zenhubusercontent.com/5c363aebf6ef6263f5aa92c3/7e325638-ea2e-48ac-ae50-d7a81fe92a8d)

Instead of getting an email like this (but should actually really look like the email template) :

![Selection_707.png](https://images.zenhubusercontent.com/5c363aebf6ef6263f5aa92c3/1297dd37-074c-401a-8e52-d3c6e0ae8f64)

Basically a naive copy of the HTML code of the new template as a string into the Lambda seems to fail. 

Next to-do's create a new Lambda function with [this documentation](https://synaptitude.atlassian.net/wiki/spaces/SYN/pages/491323393/Lambda%2Bfunctions%2Bfor%2BEmail%2Bverficiation%2BPassword%2Bupdate) as reference and attach it to synap-qa-central to debug/verify/test as to what might be going wrong.
 
References on the issue are : 

1. https://www.mihaiblaga.dev/aws-cognito-custom-email-verification-template
2. https://stackoverflow.com/questions/50804008/cognito-custom-message-with-html-tag

**Comment by aanchan** [comment]
A new template was made for email verification. The current email template looks bare. The template was designed in adobe XD. The template contained a lot of code that was not needed. The current template which is primarily HTML and CSS is attached to this.[https://app.zenhub.com/files/165302750/74cf3d08-ccb8-4d16-8c25-3b6e0012985d/download](https://app.zenhub.com/files/165302750/74cf3d08-ccb8-4d16-8c25-3b6e0012985d/download). The template looks something like this. 
![image (9).png](https://images.zenhubusercontent.com/5c363aebf6ef6263f5aa92c3/ae81c551-14e6-4fd7-8d0e-5e8f854cba92). The HTML template comes with a number of static assets. An [S3 bucket](https://s3.console.aws.amazon.com/s3/buckets/synap-confirmation-email/?region=ca-central-1) was created to hold these static assets with a cloudfront distribution (d3mseuj3wyzyi0.cloudfront.net) made to point to content.synaptitudebrainhealth.com. The equivalent HTML file is something like this :

`<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=edge"/>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Confirm Email Template</title>
<style id="applicationStylesheet" type="text/css">
	.mediaViewInfo {
		--web-view-name: Confirm Email Template;
		--web-view-id: Confirm_Email_Template;
		--web-scale-on-resize: true;
		--web-enable-deep-linking: true;
	}
	:root {
		--web-view-ids: Confirm_Email_Template;
	}
	* {
		margin: 0;
		padding: 0;
		box-sizing: border-box;
		border: none;
	}
	#Confirm_Email_Template {
		position: absolute;
		width: 800px;
		height: 1636px;
		background-color: rgba(255,255,255,1);
		overflow: hidden;
		--web-view-name: Confirm Email Template;
		--web-view-id: Confirm_Email_Template;
		--web-scale-on-resize: true;
		--web-enable-deep-linking: true;
	}
	#Welcome_to_the_Synaptitude_Bra {
		position: absolute;
		left: 139px;
		top: 239px;
		overflow: visible;
		width: 523px;
		white-space: nowrap;
		line-height: 60px;
		margin-top: -11px;
		text-align: center;
		font-family: Public Sans;
		font-style: normal;
		font-weight: bold;
		font-size: 38px;
		color: rgba(53,64,82,1);
	}
	#Group_26 {
		position: absolute;
		width: 406px;
		height: 74px;
		left: 197px;
		top: 982px;
		overflow: visible;
	}
	.Rectangle_2 {
		position: absolute;
		overflow: visible;
		width: 406px;
		height: 74px;
		left: 0px;
		top: 0px;
	}
	#Confirm_email_address {
		position: absolute;
		left: 70px;
		top: 20px;
		overflow: visible;
		width: 265px;
		white-space: nowrap;
		text-align: center;
		font-family: Roboto;
		font-style: normal;
		font-weight: bold;
		font-size: 26px;
		color: rgba(255,255,255,1);
	}
	#To_complete_your_registration_ {
		position: absolute;
		left: 109px;
		top: 813px;
		overflow: visible;
		width: 583px;
		height: 156px;
		text-align: left;
		font-family: Roboto;
		font-style: normal;
		font-weight: normal;
		font-size: 30px;
		color: rgba(53,64,82,1);
	}
	#Best_wishes__Your_brain_health {
		position: absolute;
		left: 109px;
		top: 1175px;
		overflow: visible;
		width: 306px;
		white-space: nowrap;
		line-height: 50px;
		margin-top: -10px;
		text-align: left;
		font-family: Roboto;
		font-style: normal;
		font-weight: bold;
		font-size: 30px;
		color: rgba(53,64,82,1);
	}
	#download {
		position: absolute;
		width: 137px;
		height: 93px;
		left: 332px;
		top: 62px;
		overflow: visible;
	}
	#Group_27 {
		position: absolute;
		width: 592px;
		height: 367px;
		left: 104px;
		top: 395px;
		overflow: visible;
	}
	.Rectangle_1 {
		position: absolute;
		overflow: visible;
		width: 592px;
		height: 356px;
		left: 0px;
		top: 8px;
	}
	#ID1 {
		position: absolute;
		width: 443px;
		height: 367px;
		left: 75px;
		top: 0px;
		overflow: visible;
	}
	#Group_31 {
		position: absolute;
		width: 750px;
		height: 258px;
		left: 25px;
		top: 1378px;
		overflow: visible;
	}
	.Rectangle_15 {
		position: absolute;
		overflow: visible;
		width: 750px;
		height: 258px;
		left: 0px;
		top: 0px;
	}
	#Group_29 {
		position: absolute;
		width: 712px;
		height: 228px;
		left: 19px;
		top: 24px;
		overflow: visible;
	}
	#Footer {
		position: absolute;
		width: 712px;
		height: 228px;
		left: 0px;
		top: 0px;
		overflow: visible;
	}
	#Copyright___2020_Synaptitude_B {
		position: absolute;
		left: 0px;
		top: 167px;
		overflow: visible;
		width: 713px;
		height: 61px;
		line-height: 25px;
		margin-top: -4.5px;
		text-align: center;
		font-family: Roboto;
		font-style: normal;
		font-weight: normal;
		font-size: 16px;
		color: rgba(53,64,82,1);
	}
	#Group_30 {
		position: absolute;
		width: 534px;
		height: 137px;
		left: 89px;
		top: 0px;
		overflow: visible;
	}
	#We_d_love_to_hear_from_you__Fo {
		position: absolute;
		left: 0px;
		top: 0px;
		overflow: visible;
		width: 535px;
		height: 137px;
		text-align: center;
		font-family: Roboto;
		font-style: normal;
		font-weight: bold;
		font-size: 30px;
		color: rgba(72,78,88,1);
	}
	#Symbol_1___3 {
		position: absolute;
		width: 11.339px;
		height: 21.834px;
		left: 306.492px;
		top: 123.375px;
		overflow: visible;
	}
	.facebook_2 {
		overflow: visible;
		position: absolute;
		width: 11.339px;
		height: 21.834px;
		left: 0px;
		top: 0px;
		transform: matrix(1,0,0,1,0,0);
	}
	#Rectangle_22 {
		position: absolute;
		width: 24px;
		height: 20.571px;
		left: 355.535px;
		top: 125.873px;
		overflow: visible;
	}
</style>

</head>
<body>
<div id="Confirm_Email_Template">
	<div id="Welcome_to_the_Synaptitude_Bra">
		<span>Welcome to the Synaptitude <br/>Brain Health Platform!</span>
	</div>
	<a href="http://www.synaptitudebrainhealth.com">
		<div id="Group_26">
			<svg class="Rectangle_2">
				<rect fill="rgba(93,49,201,1)" id="Rectangle_2" rx="30" ry="30" x="0" y="0" width="406" height="74">
				</rect>
			</svg>
			<div id="Confirm_email_address">
				<span>Confirm email address</span>
			</div>
		</div>
	</a>
	<div id="To_complete_your_registration_">
		<span>To complete your registration, confirm your email address with the button below: <br/></span><br>
	</div>
	<div id="Best_wishes__Your_brain_health">
		<span>Best wishes,<br/>Your brain health team</span><br>
	</div>
	<img id="download" src="http://content.synaptitudebrainhealth.com/download.png" srcset="http://content.synaptitudebrainhealth.com/download.png 1x, http://content.synaptitudebrainhealth.com/download@2x.png 2x">
	<div id="Group_27">
		<svg class="Rectangle_1">
			<rect fill="rgba(245,233,243,1)" id="Rectangle_1" rx="0" ry="0" x="0" y="0" width="592" height="356">
			</rect>
		</svg>
		<img id="ID1" src="http://content.synaptitudebrainhealth.com/ID1.png" srcset="http://content.synaptitudebrainhealth.com/ID1.png 1x, http://content.synaptitudebrainhealth.com/ID1@2x.png 2x">
	</div>
	<div id="Group_31">
		<svg class="Rectangle_15">
			<rect fill="rgba(245,245,245,1)" id="Rectangle_15" rx="0" ry="0" x="0" y="0" width="750" height="258">
			</rect>
		</svg>
		<div id="Group_29">
			<div id="Footer" class="Footer">
				<div id="Copyright___2020_Synaptitude_B">
					<span>Copyright © 2020</span><span style="font-style:normal;font-weight:bold;"> Synaptitude Brain Health</span><span>. All Rights Reserved.<br/>communications@synaptitude.info | www.synaptitudebrainhealth.com</span>
				</div>
				<div id="Group_30">
					<div id="We_d_love_to_hear_from_you__Fo">
						<span>We'd love to hear from you!</span><br><span style="font-style:normal;font-weight:lighter;font-size:20px;color:rgba(53,64,82,1);">Follow us on our Facebook Page or leave us a message. <br/>We are constantly improving and welcome feedback.</span>
					</div>
				</div>
				<div id="Symbol_1___3" class="Symbol_1___3">
					<svg class="facebook_2" viewBox="88.428 12.829 11.339 21.834">
						<path fill="rgba(53,64,82,1)" id="facebook_2" d="M 95.78785705566406 34.66301727294922 L 95.78785705566406 24.70341873168945 L 99.13092041015625 24.70341873168945 L 99.63143157958984 20.82200241088867 L 95.78785705566406 20.82200241088867 L 95.78785705566406 18.34383392333984 C 95.78785705566406 17.22009658813477 96.09995269775391 16.45420837402344 97.7115478515625 16.45420837402344 L 99.76692199707031 16.4532585144043 L 99.76692199707031 12.98177623748779 C 99.41127777099609 12.93464660644531 98.19127655029297 12.82899951934814 96.77189636230469 12.82899951934814 C 93.80850219726563 12.82899951934814 91.77970123291016 14.63775539398193 91.77970123291016 17.95962524414063 L 91.77970123291016 20.82210922241211 L 88.42799377441406 20.82210922241211 L 88.42799377441406 24.70352554321289 L 91.77960205078125 24.70352554321289 L 91.77960205078125 34.66312408447266 L 95.78785705566406 34.66301345825195 Z">
						</path>
					</svg>
				</div>
			</div>
			<img id="Rectangle_22" src="http://content.synaptitudebrainhealth.com/Rectangle_22.png" srcset="http://content.synaptitudebrainhealth.com/Rectangle_22.png 1x, http://content.synaptitudebrainhealth.com/Rectangle_22@2x.png 2x">
		</div>
	</div>
</div>
</body>
</html>`
